data/transactions/verify/txn.go

```diff
@@ -310,6 +310,15 @@ func stxnCoreChecks(gi int, groupCtx *GroupContext, batchVerifier crypto.BatchVe
 		return err
 	}
 
+	if s.Txn.Type == protocol.HeartbeatTx {
```

Comment:
I guess missing a unit test for successful and failed verification?

Reply:
I'm trying.  It's much harder to write now, assuming I want to actually run it through the whole batch verifications stxnCore stuff.  I'm adding things the txn_test.go.  Will ping you when I push the commits, then please look to confirm you agree.

Reply:
See txn_test.go
```
// TestTxnHeartbeat makes sure that a heartbeat transaction is valid (and added
// to the cache) only if the normal outer signature is valid AND the inner
// HbProof is valid.
func TestTxnHeartbeat(t *testing.T) {
```
If I understand it correctly, it tests passing and failing.